fetch.py

```diff
@@ -0,0 +1,38 @@
+import asyncio
+import logging
+import random
```

Comment:
Stylistic / subjective, but I'd recommend using `from .. import ..` over `import ...` imports. The reason- its clearer what you intend to use from this module / how reliant you are on it, and more importantly, whether your current version of the package can supply what you need.

If you try to import something on an old version of python that doesn't have that function, the `from ... import ...` will error immediately, instead of 5 minutes into running your script after you've walked away.